packages/diagrams/frontend/sirius-components-diagrams-reactflow/src/renderer/palette/DiagramPaletteToolContribution.types.ts

```diff
@@ -0,0 +1,25 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Obeo.
+ * This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v2.0
+ * which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { DiagramElementData } from './Palette.types';
+
+export interface DiagramPaletteToolContributionProps {
+  canHandle: (diagramId: string, diagramElementData: DiagramElementData) => boolean;
+  component: (props: DiagramPaletteToolContributionComponentProps) => JSX.Element | null;
```

Comment:
Shouldn't we be more strict on what this can return?
With this I can literally insert anything *inside the palette.
The tools we contribute should be able to do whatever *when invoked*, but should'nt we restrict how they appear in the palette?

Here I can literally do this:
```
        component={(props: DiagramPaletteToolContributionComponentProps) => {
          return (
            <Typography variant="h1" onClick={() => alert('whatever')}>
              wat?
            </Typography>
          );
        }}
```
and get 
![image](https://github.com/eclipse-sirius/sirius-web/assets/10608/8579393e-7882-4b3a-992e-d88d2380275f)
as a "palette entry".

Reply:
After discussion with @sbegaudeau, we don't add restrictive controls and let the developper contributes whatever he wants.